DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java

```diff
@@ -0,0 +1,168 @@
+package com.rfs;
+
+import com.rfs.cms.ApacheHttpClient;
+import com.rfs.cms.OpenSearchWorkCoordinator;
+import com.rfs.cms.ProcessManager;
+import com.rfs.common.ClusterVersion;
+import com.rfs.common.DefaultSourceRepoAccessor;
+import com.rfs.common.DocumentReindexer;
+import com.rfs.common.FileSystemRepo;
+import com.rfs.common.FileSystemSnapshotCreator;
+import com.rfs.common.GlobalMetadata;
+import com.rfs.common.IndexMetadata;
+import com.rfs.common.LuceneDocumentsReader;
+import com.rfs.common.OpenSearchClient;
+import com.rfs.common.ShardMetadata;
+import com.rfs.common.SnapshotRepo;
+import com.rfs.common.SnapshotShardUnpacker;
+import com.rfs.common.SourceRepo;
+import com.rfs.framework.ElasticsearchContainer;
+import com.rfs.transformers.TransformFunctions;
+import com.rfs.transformers.Transformer;
+import com.rfs.version_es_7_10.ElasticsearchConstants_ES_7_10;
+import com.rfs.version_es_7_10.GlobalMetadataFactory_ES_7_10;
+import com.rfs.version_es_7_10.IndexMetadataFactory_ES_7_10;
+import com.rfs.version_es_7_10.ShardMetadataFactory_ES_7_10;
+import com.rfs.version_es_7_10.SnapshotRepoProvider_ES_7_10;
+import com.rfs.version_os_2_11.GlobalMetadataCreator_OS_2_11;
+import com.rfs.version_os_2_11.IndexCreator_OS_2_11;
+import com.rfs.worker.IndexRunner;
+import com.rfs.worker.MetadataRunner;
+import lombok.Lombok;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.lucene.document.Document;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.opensearch.testcontainers.OpensearchContainer;
+import reactor.core.publisher.Flux;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.UnaryOperator;
+
+@Tag("longTest")
+@Slf4j
+public class FullTest {
+    final static long TOLERABLE_CLIENT_SERVER_CLOCK_DIFFERENCE_SECONDS = 3600;
+    final static ElasticsearchContainer esSourceContainer =
+            new ElasticsearchContainer(new ElasticsearchContainer.Version("elasticsearch_rfs_source",
+                    "preloaded-ES_7_10"));
+    final static OpensearchContainer<?> osTargetContainer =
+            new OpensearchContainer<>("opensearchproject/opensearch:2.13.0");
+
+    @BeforeAll
+    static void setupSourceContainer() {
+        esSourceContainer.start();
```

Comment:
Think you need to put some things in the source container to expect to be migrated

Reply:
The source container is the elasticsearch_rfs_source container, which is being built automatically by gradle (every time still :( ).  For now, that includes our 4 OpenSearch Benchmark test runs of 1000 docs / index.  I'd like to improve our cacheable image generation and to include different variants (7.17, 6.8; many more shards; fewer shards), etc.

That will probably be in future PRs though.